src/popup/App.tsx

```diff
@@ -20,127 +20,145 @@ import emptyInbox from './assets/empty_inbox.svg';
-                        return <MergeRequest mr={mr} key={mr.id} />;
-                    });
+    // call fetch data at component mount
+    useEffect(() => fetchData(), [fetchData]);
 
-                    updateList(<FilterList className={'mrList'}>{listItems}</FilterList>);
-                }
+    const getContent = useCallback(() => {
+        if (appStatus === 'idle') {
+            return <Text>Fetching content...</Text>;
+        }
 
-                setMrToReview(`${response.mrToReview}`);
-                setMrReviewed(`${response.mrReviewed} / ${response.mrGiven.length}`);
-                setMrRatio(Math.floor(((mrNewList.length - response.mrToReview) / mrNewList.length) * 100) || 100);
-                setLastUpdateDateUnix(response.lastUpdateDateUnix);
-            })
-            .catch((error) => console.error(error));
-    }, []);
+        if (appStatus === 'loading') {
+            return <Text>Fetching content...</Text>;
+        }
 
-    if (mrList === fetchingContentState) {
-        sendMsg(null);
-    }
+        if (appStatus === 'error') {
+            return (
+                <Flash m={2} scheme="red">
+                    {errorMessage}
+                </Flash>
+            );
+        }
+
+        // show data
+        let mrList = [];
+        if (currentTab === 0) {
+            mrList = mrData.mrAssigned;
+        } else {
+            mrList = mrData.mrGiven;
+        }
+
+        if (!mrList || mrList.length === 0) {
+            return <img src={emptyInbox} className={'emptyInbox'} />;
+        }
+        return (
+            <FilterList className={'mrList'}>
+                {mrList.map((mr: MergeRequestsDetails) => (
+                    <MergeRequest mr={mr} key={mr.id} />
+                ))}
+            </FilterList>
+        );
+    }, [appStatus, mrData, currentTab, errorMessage]);
+
+    const getMrRatio = useCallback(() => {
+        if (appStatus !== 'success' || mrData === null) {
+            return 100;
+        }
+        let mrList = [];
+        if (currentTab === 0) {
+            mrList = mrData.mrAssigned;
+        } else {
+            mrList = mrData.mrGiven;
+        }
+        let rate = 0;
+        if (mrList && mrList.length > 0) {
+            rate = (mrList.length - mrData.mrToReview) / mrList.length;
+        }
+        return Math.floor(rate * 100);
+    }, [appStatus, mrData, currentTab]);
 
     return (
         <ThemeProvider theme={primer}>
             <div className={'container'}>
                 <TabNav aria-label="Main" mb={2}>
-                    <TabNav.Link
-                        onClick={sendMsg}
-                        data-key={'getMRs'}
-                        data-id={'0'}
-                        href="#MR"
-                        className={tabSelected === '0' ? 'selected' : ''}
-                    >
+                    <TabNav.Link onClick={() => setCurrentTab(0)} className={currentTab === 0 ? 'selected' : ''}>
```

Comment:
I'm not sure why, but we lost the "pointer" mouse when hovering the tab menu.

Reply:
is it for the href?

Reply:
ah yes, so maybe add a bit of CSS, otherwise it's less intuitive to click on it ^^